src/rp2_common/pico_btstack/CMakeLists.txt

```diff
@@ -181,9 +193,11 @@ if (EXISTS ${PICO_BTSTACK_PATH}/${BTSTACK_TEST_PATH})
             ${PICO_BTSTACK_PATH}/3rd-party/bluedroid/encoder/srce/sbc_encoder.c
             ${PICO_BTSTACK_PATH}/3rd-party/bluedroid/encoder/srce/sbc_packing.c
             )
-    target_include_directories(pico_btstack_sbc_encoder_headers INTERFACE
-            ${PICO_BTSTACK_PATH}/3rd-party/bluedroid/encoder/include
-            )
+    if (PICO_BTSTACK_VERSION VERSION_GREATER_EQUAL "1.6.0")
```

Comment:
If you like, you can drop this check (less complexity). The SBC codec has been around for many years in BTstack

Reply:
The check for version 1.6.1 is just whether to add the source file btstack_sbc_bluedroid.c which seems to be new.

I haven't been taking much care updating this cmake file in the past and someone was complaining that it can cause problems if they have their own copy of btstack using an older version.

Reply:
Well, it isn't. The SBC codec in 3rd-part/bluedroid was added in 2016 which pre-dates pico-sdk by a few years.
(BTstack v1.1 was tagged in 2019 btw).

So, I'd suggest to remove this check (to avoid clutter) and ignore or discuss with the person that complained about it.

Reply:
btstack_sbc_bluedroid.c was added in Nov 2023. We last updated in June 2023. If I don't add it I get a link error for btstack_sbc_encoder_bluedroid_init_instance. If I add it without any of this clutter older versions of btstack won't build. This is the issue I was thinking about https://github.com/raspberrypi/pico-sdk/issues/1553

```
commit 113ff2bc0af39c4454c5c239318755ec65070fce
Author: Matthias Ringwald <matthias@ringwald.ch>
Date:   Wed Nov 8 15:06:21 2023 +0100

    btstack_sbc_bluedroid: implement encoder for Bluedroid SBC

```

Reply:
Oh, sorry, my bad. The Bluedroid codec (incl. headers) has 'always' been there, there was a need to refactor how it's used and that's the new file. Ok, go ahead with the checks (which are not clutter then :)
On the other hand, one could argue that people should not downgrade BTstack within Pico SDK.

Reply:
> people should not downgrade BTstack within Pico SDK

I agree. The issue here is we allow clients to use their own btstack folder in their project. So it's not that they are downgrading, just that they haven't (yet) upgraded.